---
layout: notebook_resource_template
title: IGF help pages - Notebook Resources
---

# Notebook resources

## List of notebook resources

A list of all the notebooks used by IGF for generating QC plots and analysis results can be found here.

<div class="table-responsive">
<table class="table table-hover">
  <thead class="table-light">
    <tr style="text-align: center;">
      <th scope="col">Analysis name</th>
      <th scope="col">Documentation</th>
      <th scope="col">Example notebooks</th>
      <th scope="col">Binder</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>Singlecell data analysis using Scanpy</td>
      <td><a href="https://github.com/imperial-genomics-facility/scanpy-notebook-image">Github docs</a></td>
      <td><ul><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/scanpy-notebook-image/blob/master/examples/Clustering_3K_PBMCs_v0.0.4.1.ipynb">Scanpy - Clustering of 3K PBMC cells</a></li></ul></td>
      <td><ul><li><a href="https://mybinder.org/v2/gh/imperial-genomics-facility/scanpy-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2FClustering_3K_PBMCs_v0.0.4.1.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li></ul></td>
    </tr>
    <tr>
      <td>Singlecell data analysis using Seurat (dev)</td>
      <td><a href="https://seurat-notebook-image.readthedocs.io/en/latest/">Help docs</a></td>
      <td><ul><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/seurat-notebook-image/blob/master/examples/seurat_basic_v0.0.1.ipynb">Seurat - Clustering of 3k PBMC cells</a></li></ul></td>
      <td><ul><li><a href="https://mybinder.org/v2/gh/imperial-genomics-facility/seurat-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2Fseurat_basic_v0.0.1.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li></ul></td>
    </tr>
    <tr>
      <td>RNA-Seq data analysis (dev)</td>
      <td><a href="https://rnaseq-notebook-image.readthedocs.io/en/latest/">Help docs</a></td>
      <td><ul><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/rnaseq-notebook-image/blob/master/examples/rnaseq_example_1_v0.0.1.ipynb">RNA-Seq data using STAR and DESeq2</a></li><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/rnaseq-notebook-image/blob/master/examples/rnaseq_example_dask_1_v0.0.1.ipynb">RNA-Seq data using STAR and DESeq2 (using Dask)</a></li></ul></td>
      <td><ul><li><a href="https://notebooks.gesis.org/binder/v2/gh/imperial-genomics-facility/rnaseq-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2Frnaseq_example_1_v0.0.1.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li><li><a href="https://notebooks.gesis.org/binder/v2/gh/imperial-genomics-facility/rnaseq-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2Frnaseq_example_dask_1_v0.0.1.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li></ul></td>
    </tr>
    <tr>
      <td>Single cell TCR/RNA-seq data analysis using Scirpy (dev)</td>
      <td><a href="https://github.com/imperial-genomics-facility/scirpy-notebook-image">Github docs</a></td>
      <td><ul><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/scirpy-notebook-image/blob/main/examples/Scirpy_tutorial_3k_tcr.ipynb">Analysis of 3k T cells from cancer</a></li></ul></td>
      <td><ul><li><a href="https://mybinder.org/v2/gh/imperial-genomics-facility/scirpy-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2FScirpy_tutorial_3k_tcr.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li></ul></td>
    </tr>
    <tr>
      <td>GWAS data analysis using Hail (dev)</td>
      <td><a href="https://hail-notebook-image.readthedocs.io/en/latest">Help docs</a></td>
      <td><ul><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/hail-notebook-image/blob/master/examples/hail_GWAS_v0.0.1.ipynb">Hail - GWAS Tutorial</a></li></ul></td>
      <td><ul><li><a href="https://mybinder.org/v2/gh/imperial-genomics-facility/hail-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2Fhail_GWAS_v0.0.1.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li></ul></td>
    </tr>
    <tr>
      <td>Viral genome analysis (dev)</td>
      <td><a href="https://viral-genome-notebook-image.readthedocs.io/en/latest/">Help docs</a></td>
      <td><ul><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/viral-genome-notebook-image/blob/master/examples/coronavirus_analysis_build_consensus_fasta.ipynb">Viral genome - building consensus genome</a></li><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/viral-genome-notebook-image/blob/master/examples/coronavirus_analysis_multiple_sequence_alignment_and_tree_building.ipynb">Viral genome - MSA and phylogenetic tree</a></li><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/viral-genome-notebook-image/blob/master/examples/coronavirus_analysis_Tree_building_for_153_Coronavirus_genomes.ipynb">Coronavirus - Phylogenetic tree of 153 genomes</a></li><li><a href="https://nbviewer.jupyter.org/github/imperial-genomics-facility/viral-genome-notebook-image/blob/master/examples/coronavirus_analysis_alignment_and_tree_building_using_Nextstrain.ipynb">Viral genome analysis using Nextstrain</a></li></ul></td>
      <td><ul><li><a href="https://mybinder.org/v2/gh/imperial-genomics-facility/viral-genome-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2Fcoronavirus_analysis_build_consensus_fasta.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li><li><a href="https://mybinder.org/v2/gh/imperial-genomics-facility/viral-genome-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2Fcoronavirus_analysis_multiple_sequence_alignment_and_tree_building.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li><li><a href="https://mybinder.org/v2/gh/imperial-genomics-facility/viral-genome-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2Fcoronavirus_analysis_Tree_building_for_153_Coronavirus_genomes.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li><li><a href="https://mybinder.org/v2/gh/imperial-genomics-facility/viral-genome-notebook-image/master?urlpath=lab%2Ftree%2Fexamples%2Fcoronavirus_analysis_alignment_and_tree_building_using_Nextstrain.ipynb"><img alt="Binder" src="https://mybinder.org/badge_logo.svg"></a></li></ul></td>
    </tr>
  </tbody>
</table>
</div>

__Note:__ 
* A laptop / desktop with any modern web browser is required to explore these notebooks
* Binder provides a temporary environment for running these example notebooks and any changes to the notebook will get deleted after a while
* Start up time may vary for these notebooks as Binder needs to build a new environment once there are any new changes in these repositories
* Repositories with tag `dev` are still under active development and the notebook runs may fail occasionally 
* All these notebooks will have `Python`/`R`/`bash` scripts. Although you can run these notebooks without knowing any of these programming languages, its recommended to have some basic knowledge, if you are planning to edit the code

### Where to run?

You can run these notebooks on

* __Binder__: Load these example notebooks using limited, free computational resource (using example data)
* __Docker__ (on your Laptop/Desktop): You will have to install Docker [(https://docs.docker.com/get-docker/)](https://docs.docker.com/get-docker/) on your own laptop or server, if you prefer to check your own data
* __Imperial College HPC__: You have to install all the tools on a conda environment and then you can copy and run these notebooks on the HPC. More details on how to set up conda environment can be found here: [https://www.imperial.ac.uk/admin-services/ict/self-service/research-support/rcs/support/applications/jupyter/](https://www.imperial.ac.uk/admin-services/ict/self-service/research-support/rcs/support/applications/jupyter/)


__Note__: Installing Docker will require root (admin) access on your laptop or server and you may need to contact the IT admin to complete this step.

__Warning__: Docker images downloaded from unknown sources can have malicious codes which can damage your laptop or server.

## Change logs
* None
